test/custom-resource-handlers/private-key.test.ts

```diff
@@ -0,0 +1,185 @@
+jest.spyOn(aws, 'SecretsManager').mockImplementation(() => mockSecretsManager);
+mockSecretsManager.createSecret = jest.fn().mockName('SecretsManager.createSecret')
+  .mockImplementation(() => ({ promise: () => Promise.resolve({ ARN: secretArn, VersionId: secretVersionId }) })) as any;
+mockSecretsManager.updateSecret = jest.fn().mockName('SecretsManager.updateSecret')
+  .mockImplementation(() => ({ promise: () => Promise.resolve({ ARN: secretArn }) })) as any;
+mockSecretsManager.deleteSecret = jest.fn().mockName('SecretsManager.deleteSecret')
+  .mockImplementation(() => ({ promise: () => Promise.resolve({}) })) as any;
+const mockRmrf = jest.fn().mockName('_rmrf').mockResolvedValue(undefined);
+jest.mock('../../custom-resource-handlers/src/_rmrf', () => mockRmrf);
+secretsManager.resolveCurrentVersionId = jest.fn().mockName('resolveCurrentVersionId')
+  .mockResolvedValue(secretVersionId);
+
+beforeEach(() => jest.clearAllMocks());
+
+test('Create', async () => {
+  const event: cfn.Event = {
+    RequestType: cfn.RequestType.CREATE,
+    PhysicalResourceId: undefined,
+    ...eventBase,
+  };
+
+  const { main } = require('../../custom-resource-handlers/src/private-key');
+  await expect(main(event, context)).resolves.toBe(undefined);
+
+  await expect(mockSecretsManager.createSecret)
+    .toBeCalledWith({
+      ClientRequestToken: context.awsRequestId,
+      Description: event.ResourceProperties.Description,
+      KmsKeyId: event.ResourceProperties.KmsKeyId,
+      Name: event.ResourceProperties.SecretName,
+      SecretString: mockPrivateKey,
+    });
+  await expect(mockSecretsManager.updateSecret).not.toBeCalled();
+  await expect(mockSecretsManager.deleteSecret).not.toBeCalled();
+  await expect(mockRmrf).toBeCalledWith(mockTmpDir);
+  return expect(cfn.sendResponse)
+    .toBeCalledWith(event,
+                    cfn.Status.SUCCESS,
+                    secretArn,
+                    { SecretArn: secretArn, SecretVersionId: secretVersionId });
+});
+
+test('Update (changing KeySize)', async () => {
+  const event: cfn.Event = {
+    RequestType: cfn.RequestType.UPDATE,
+    PhysicalResourceId: secretArn,
+    OldResourceProperties: {
+      ...eventBase.ResourceProperties,
+      KeySize: mockKeySize * 2,
+    },
+    ...eventBase,
+  };
+
+  const { main } = require('../../custom-resource-handlers/src/private-key');
+  await expect(main(event, context)).resolves.toBe(undefined);
+
+  await expect(mockSecretsManager.createSecret).not.toBeCalled();
+  await expect(mockSecretsManager.updateSecret).not.toBeCalled();
+  await expect(mockSecretsManager.deleteSecret).not.toBeCalled();
+  return expect(cfn.sendResponse)
+    .toBeCalledWith(event,
+                    cfn.Status.FAILED,
+                    secretArn,
+                    { SecretArn: '' },
+                    expect.stringContaining('The KeySize property cannot be updated'));
+});
+
+test('Update (changing KeySize)', async () => {
```

Comment:
Looks like a dup test

Reply:
That is what you get for having your lunch break right after copying a test with intention of modifying it for a variant (but before having modified it).